draft-ietf-jsonpath-base.md

```diff
@@ -1264,6 +1264,32 @@ Queries:
 | `$[0, 0]` | `"a"` <br> `"a"` | `$[0]` <br> `$[0]` | Duplicated entries |
 {: title="List selector examples"}
 
+## Semantics of `null`
+
+Note that JSON `null` is treated the same as any other JSON value: it is not taken to mean "undefined" or "missing".
+
+### Examples
+{: unnumbered}
+
+JSON document:
+
+    {"a": null, "b": [null], "c": [{}], "null": 1}
+
+Queries:
+
+| Query | Result | Result Paths | Comment |
+| :---: | ------ | :----------: | ------- |
+| `$.a` | `null` | `$['a']` | Object value |
+| `$.a[0]` | | | `null` used as array |
+| `$.a.d` | | | `null` used as object |
+| `$.b[0]` | `null` | `$['b'][0]` | Array value |
+| `$.b[*]` | `null` | `$['b'][0]` | Array value |
+| `$.b[?@]` | `null` | `$['b'][0]` | Existence |
+| `$.b[?@==null]` | `null` | `$['b'][0]` | Comparison |
```

Comment:
I believe these need parentheses

```suggestion
| `$.b[?(@)]` | `null` | `$['b'][0]` | Existence |
| `$.b[?(@==null)]` | `null` | `$['b'][0]` | Comparison |
```

We had discussed making them optional somewhere, but I don't think we elected to allow it.

Reply:
@gregsdennis These examples merely follow the [current ABNF](https://ietf-wg-jsonpath.github.io/draft-ietf-jsonpath-base/draft-ietf-jsonpath-base.html#name-filter-selector) in which those parentheses are still optional. Happy to add the parentheses if we agree the ABNF should be changed to mandate the parentheses, but I think it would be best to discuss that separately to this PR (which may have a rather small audience).

Reply:
If that's the case, the ABNF needs to be fixed.  We never agreed to allow these parentheses to be omitted.

The closest related discussion on this I could find is https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/92#issuecomment-809331011 and https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/92#issuecomment-809601525.  These comments clearly indicate that the parentheses _should_ be present.

Reply:
@gregsdennis  I think you should take this up on the mailing list or via a new issue.

Reply:
@gregsdennis If you are comfortable with these examples reflecting the current spec, please approve so I can merge.